### PR TITLE
Add developer documentation

### LINKED ISSUE #35: Add developer documentation
### Motivation

As long as there are no application-related issues, we could also start to set up a developer documentation, so we can gradually add information down the way and don't have a huge workload when we want to add a documentation later on.

### Proposed Solution

- Add the following dev dependencies:
  - "sphinx"
  - "sphinxcontrib-django"
  - "sphinx-last-updated-by-git"
  - "sphinx-rtd-theme"
- Add a `docs` directory
- Add basic sphinx configuration (can be inspired by the `conf.py` of [integreat-cms](https://github.com/digitalfabrik/integreat-cms/blob/develop/docs/src/conf.py) or [lunes-cms](https://github.com/digitalfabrik/lunes-cms/blob/develop/docs/src/conf.py))
- Add a script to build the documentation locally, see e.g. [build_documentation.sh](https://github.com/digitalfabrik/lunes-cms/blob/develop/tools/build_documentation.sh) or [make_docs.sh](https://github.com/digitalfabrik/integreat-cms/blob/develop/tools/make_docs.sh)
- Decide how the documentation is published... either write a deployment script for github pages and add it to the circleci workflow (like [integreat-cms](https://digitalfabrik.github.io/integreat-cms/)), or sign up on https://readthedocs.org/ like [lunes-cms](https://lunes-cms.readthedocs.io/en/latest/?badge=latest) (which could be the easier solution here and also has the advantage that all tagged versions can be viewed in the documentation, not only the latest version)
  - Read the docs config can be copied from lunes: [.readthedocs.yaml](https://github.com/digitalfabrik/lunes-cms/blob/develop/.readthedocs.yaml)

### Alternatives

A developer documentation could also be out of scope for this project, since it also takes time resources away from normal developing time and given the limited scope, the relevant information could also be contained in the inline documentation or a bunch of `.md` files

### PR DESCRIPTION
**Pull request by charludo**
### Short description

Add developer documentation. The documentation and configs are largely copied from `lunes-cms`

### Proposed changes

- add `make_docs.sh` script and sphinx config files
- add developer documentation 
- added translation dev tools (since this PR was already adding documentation for them...)

### Side effects

- I do not think I have sufficient permissions for RTD to create its webhook, so there's no automated builds of any kind ATM.

### Resolved issues

Fixes: #35

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# .readthedocs.yaml
+# Read the Docs configuration file
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+version: 2
+
+build:
+  os: ubuntu-20.04
```
Comment (timobrembeck) [inline .readthedocs.yaml]:
```suggestion
  os: ubuntu-22.04
```
Maybe go with the newer LTS version?

File: docs/src/index.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+.. toctree::
+    :caption: Deployment
+    :hidden:
+
+    packaging
+    prod-server
+    changelog
+
+* :doc:`packaging`: Create an easy installable python
```
Comment (timobrembeck) [inline docs/src/index.rst]:
```suggestion
* :doc:`packaging`: Create an easy installable python package
```

File: pyproject.toml [hunk 3]
```diff
@@ -44,6 +44,10 @@ dev = [
     "pytest-testmon",
     "pytest-xdist",
     "pytest-icdiff",
+    "sphinx",
+    "sphinxcontrib-django",
+    "sphinx-last-updated-by-git",
+    "sphinx-rtd-theme",
```
Comment (timobrembeck) [inline pyproject.toml:53]: Could you execute `./tools/update_dependencies.sh --only=pip` to add these dependencies to the pinned versions as well?

File: .readthedocs.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  jobs:
+    pre_build:
+      - ./tools/make_docs.sh
+
+python:
+  install:
+    - method: pip
+      path: .
+      extra_requirements:
+        - doc
```
Comment (timobrembeck) [inline .readthedocs.yaml]:
```suggestion
        - pinned
        - dev-pinned
```

**Review by timobrembeck — CHANGES_REQUESTED** [review]
Great, thanks! :100: 

Just a few config problems remain I think.

**Review by timobrembeck — CHANGES_REQUESTED** [review]
Thanks for your changes!
I added the GitHub webhook integration, so pushes on develop and PRs are now built automatically.

The build for this PR however is still failing, maybe you could have a look?
https://readthedocs.org/projects/integreat-compass/builds/20245796/
Screenshot 2023-04-26 at 12-47-36 integreat-compass Read the Docs

Once the build works, the preview of this PR would be available at https://integreat-compass--44.org.readthedocs.build/en/44/

**Comment by charludo** [comment]
@timoludwig it should be working now: [https://integreat-compass.readthedocs.io/en/add-developer-documentation/index.html](https://integreat-compass.readthedocs.io/en/add-developer-documentation/index.html)

The environment variable `INTEGREAT_COMPASS_INSTALLED`was missing.

Edit: should be back in 5-6 minutes... I don´t know how you built for this branch specifically? I activated builds for this branch in general, but deactivating it again has wiped the documentation

**Comment by timobrembeck** [comment]
> Edit: should be back in 5-6 minutes... I don´t know how you built for this branch specifically? I activated builds for this branch in general, but deactivating it again has wiped the documentation

Pushing to this branch should automatically trigger the PR build again...
Could you resolve the conflicts and squash the commits?

**Comment by timobrembeck** [comment]
This looks cool! https://integreat-compass--44.org.readthedocs.build/en/44/

Could you squash the two commits?

**Comment by charludo** [comment]
Oh whoops, of course.

**Review by timobrembeck — APPROVED** [review]
:rocket:

**Pull request merged**